tests/snippets/floats.py

```diff
@@ -98,6 +98,18 @@
 assert_raises(ValueError, lambda: float('foo'))
 assert_raises(OverflowError, lambda: float(2**10000))
 
+# check eq and hash for small numbers
+
+assert 1.0 == 1
+assert 1.0 == True
+assert 0.0 == 0
+assert 0.0 == False
+assert hash(1.0) == hash(1)
+assert hash(1.0) == hash(True)
+assert hash(0.0) == hash(0)
+assert hash(0.0) == hash(False)
+assert hash(1.0) != hash(1.0000000001)
```

Comment:
could we also get a "real world" test case, for example this?
```
assert 5.0 in {3, 4, 5}
```

Reply:
Added a few cases. The test added in dict.py also depends on hash